addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/BaseChannelHandler.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.lgwebos.internal;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.openhab.binding.lgwebos.handler.LGWebOSHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.connectsdk.device.ConnectableDevice;
+import com.connectsdk.service.capability.listeners.ResponseListener;
+import com.connectsdk.service.command.ServiceCommandError;
+import com.connectsdk.service.command.ServiceSubscription;
+
+/**
+ * An abstract implementation of ChannelHander which serves as a base class for all concrete instances.
+ *
+ * @author Sebastian Prehn
+ * @since 1.8.0
+ */
+abstract class BaseChannelHandler<T> implements ChannelHandler {
+    private Logger logger = LoggerFactory.getLogger(BaseChannelHandler.class);
+
+    // IP to Subscriptions map
+    private Map<String, ServiceSubscription<T>> subscriptions;
+
+    // lazy init
```

Comment:
Why, I see no benefit?

Reply:
Not all channels support subscriptions. So I figured it would not be good to allocate a hashmap for every channel if it is never used in some cases

Reply:
I'll change return type of getSubscription to Optional to make this clearer.

Reply:
I believe we are not using `Optional` in this project but stick to `null`s and handling them were needed with some help of the annotations.

Reply:
I would like to leave it as is now as it  is standard Java8 and it is actually required to work with optionals when working with the stream and findFirst approach. After all, it is completely binding internal.